### PR TITLE
Notes about Node16 migration

### PR DESCRIPTION
**Pull request by AndreyIvanov42**
## Task description

To migrate to node 16 some changes are needed to make for tasks. This PR contains information about it.

## Changelog

- Added section "Upgrading Tasks to Node 16"
- Added section "Common packages dependent on `azure-pipeline-task-lib` , `azure-pipeline-tool-lib`"
- Added section "List of known dependency issues"
- Added section "Feedback"

### CODE REVIEW

**Review by max-zaytsev — APPROVED** [review]

File: docs/migrateNode16.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+1. Update @types packages in `package.json` dependencies.
+
+```json
+  "dependencies": {
+    "@types/node": "^16.11.39",
+    "@types/mocha": "^9.1.1",
+    ...
+  }
+```
+2. Upgrade `azure-pipelines-task-lib` to `^4.0.0` in package.json dependencies.
```
Comment (alexander-smolyakov) [inline docs/migrateNode16.md:18]: Please add a note that at the moment related packages are in the preview state.

File: docs/migrateNode16.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
+# Table of content
+- [Table of content](#table-of-content)
+  - [Upgrading Tasks to Node 16](#upgrading-tasks-to-node-16)
+  - [Common packages dependent on `azure-pipeline-task-lib`](#common-packages-dependent-on-azure-pipeline-task-lib)
+  - [List of known dependency issues](#list-of-known-dependency-issues)
+
+## Upgrading Tasks to Node 16
```
Comment (alexander-smolyakov) [inline docs/migrateNode16.md:7]: Please add information about how customers can provide feedback on preview packages.

File: docs/migrateNode16.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
+# Table of content
+- [Table of content](#table-of-content)
+  - [Upgrading Tasks to Node 16](#upgrading-tasks-to-node-16)
```
Comment (alexander-smolyakov) [inline docs/migrateNode16.md:3]: Let's also add info about task-tool-lib

File: docs/migrateNode16.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+**Node 17:**
+fs:
+- [6cd12be347] - (SEMVER-MINOR) fs: add FileHandle.prototype.readableWebStream() (James M Snell) #39331
+
+**Node 18:**
+fs
+- (SEMVER-MAJOR) fs: runtime deprecate string coercion in fs.write, fs.writeFileSync (Livia Medeiros) #42607
+- (SEMVER-MAJOR) child_process: improve argument validation (Rich Trott) #41305
+- doc: add RafaelGSS to collaborators (RafaelGSS) #42718
```
Comment (alexander-smolyakov) [inline docs/migrateNode16.md:135]: let's remove this info since not actual at the moment

**Review by alexander-smolyakov — COMMENTED** [review]
Please take a look at the comments

**Review by kirill-ivlev — APPROVED** [review]

**Review by wojgasior — APPROVED** [review]

**Pull request merged**